### DISCUSSION TITLE
Generic arguments with range operator?

### DISCUSSION THREAD

**Discussion by tacosontitan** [dotnet/csharplang#5078] [Language Ideas]
I'm sure this is going to be a bit of a niche topic, and likely too much work for too little reward, but I *think* it would be nice to have the ability to utilize the range operator `..` with generic arguments:

    public abstract class DataHandler<TIn, TOut> { ... }
    public sealed class EntityDataHandler<Entity..2> { ... }

I believe the syntax may need to change (due to lack of sleep on my end), but currently, I *think* it would behave as:

> Given a generic type argument T, the range operator will apply the specified type starting with Tn and ending with Tn + i.

So given a list of `T1, T2, ... T9`, if the range operator is specified as follows:

    <int, string, bool..8, float>

Then the types used for `T3` through `T8` are of type `bool`.

**Comment by ufcpp** [comment]
https://github.com/dotnet/csharplang/discussions/2330
https://github.com/dotnet/csharplang/discussions/4415
